src/snowflake/snowpark/_internal/server_connection.py

```diff
@@ -426,16 +432,20 @@ def get_result_set(
         if result is None:
             raise SnowparkClientExceptionMessages.SQL_LAST_QUERY_RETURN_RESULTSET()
 
-        return result["data"], result_meta
+        return result, result_meta
 
     def get_result_and_metadata(
         self, plan: SnowflakePlan, **kwargs
-    ) -> Union[List[Row], List[Attribute]]:
+    ) -> Tuple[List[Row], List[Attribute]]:
         result_set, result_meta = self.get_result_set(plan, **kwargs)
-        result = result_set_to_rows(result_set)
+        result = result_set_to_rows(result_set["data"])
         meta = convert_result_meta_to_attribute(result_meta)
         return result, meta
 
+    def get_result_query_id(self, plan: SnowflakePlan, **kwargs) -> str:
+        result_set, _ = self.get_result_set(plan, **kwargs)
```

Comment:
@sfc-gh-jdu I just realized whether this would fetch the whole result set to the client. If so this may use a lot of memory. So does it fetch data?

Reply:
Yeah it does, I will put out another PR that this method only gets the iterator but doesn't fetch it.